src/snowflake/snowpark/dataframe.py

```diff
@@ -1089,6 +1091,32 @@ def col(self, col_name: str) -> Column:
         else:
             return Column(self._resolve(col_name))
 
+    def col_regex(self, regex: str, case_sensitive: bool = False) -> Column:
```

Comment:
I think regex match should be case sensitive by default, because that is default behavior of re.
-> `case_sensitive: bool = True`

Reply:
we make case-insensitive as default because snowflake column name by default is case-insensitive. snowflake will add double quotes and all-CAP them if you don't quote them in the advance

Reply:
lets also take Yijun's opinion on this.